detectable/src/main/java/com/synopsys/integration/detectable/detectables/bazel/BazelDetectableOptions.java

```diff
@@ -22,30 +22,62 @@
  */
 package com.synopsys.integration.detectable.detectables.bazel;
 
+import java.util.Arrays;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Optional;
+import java.util.Set;
+
+import com.synopsys.integration.configuration.property.types.enumfilterable.FilterableEnumUtils;
+import com.synopsys.integration.configuration.property.types.enumfilterable.FilterableEnumValue;
 
 public class BazelDetectableOptions {
     private final String targetName;
-    private final WorkspaceRule bazelDependencyRule;
+    private final List<FilterableEnumValue<WorkspaceRule>> bazelDependencyRulesPropertyValues;
     private final List<String> bazelCqueryAdditionalOptions;
 
-    public BazelDetectableOptions(final String targetName, final WorkspaceRule bazelDependencyRule,
-        final List<String> bazelCqueryAdditionalOptions) {
+    public BazelDetectableOptions(String targetName, List<FilterableEnumValue<WorkspaceRule>> bazelDependencyRulesPropertyValues,
+        List<String> bazelCqueryAdditionalOptions) {
         this.targetName = targetName;
-        this.bazelDependencyRule = bazelDependencyRule;
+        this.bazelDependencyRulesPropertyValues = bazelDependencyRulesPropertyValues;
         this.bazelCqueryAdditionalOptions = bazelCqueryAdditionalOptions;
     }
 
     public Optional<String> getTargetName() {
         return Optional.ofNullable(targetName);
     }
 
-    public WorkspaceRule getBazelDependencyRule() {
-        return bazelDependencyRule;
-    }
-
     public List<String> getBazelCqueryAdditionalOptions() {
         return bazelCqueryAdditionalOptions;
     }
+
+    public Set<WorkspaceRule> getBazelDependencyRules() {
```

Comment:
I think this work should move to the configuration factory - imo options shouldn't have logic, they are POJO and should be constructed in a valid state. This would let detectable not need to know about FilterableEnumValue - the configuration just converts that to the appropriate Set<WorkspaceRule> and creates a valid option object.

Reply:
I moved it to DetectableOptionFactory... is that what you meant?